File: Stack/Stack_by_linked_list/stack.java
package Stack_by_linked_list;

class stacknode {
    public int data;
    public stacknode next;

    public stacknode(int data) {
        this.data = data;
        this.next = null;
    }

    public int getData() {
        return this.data;
    }

    public void setData(int data) {
        this.data = data;
    }

    public stacknode getNext() {
        return this.next;
    }

    public void setNext(stacknode next) {
        this.next = next;
    }

    public boolean isEmpty(stacknode node) {
        return node == null;
    }

    public void push(stacknode node, int data) {
        if (isEmpty(node)) {
            return;
        }
        stacknode newnode = new stacknode(data);
        stacknode temp = node;
        newnode.next = temp;
        node = newnode;

        System.out.println(data + " pushed to stack .");
    }

    public int pop(stacknode node) {
        if (isEmpty(node)) {
            System.out.println("Stack is underflow !!!");
            return 0;
        }
        stacknode temp = node;
        node = node.next;
        int popped = temp.data;

        return popped;
    }

    public int peek(stacknode node) {
        if (isEmpty(node)) {
            System.out.println("Stack is underflow !!!");
            return 0;
        }
        return node.data;
    }
}

public class stack {
    public static void main(String[] args) {
        stacknode stack = new stacknode(10);
        stack.push(stack, 30);
        stack.push(stack, 40);
        stack.push(stack, 50);

        System.out.println("Top element is : " + stack.peek(stack));
    }
}

File: Stack/Stack_by_linked_list/stack.cs
using System ; 
namespace Stack.Stack_by_linked_list{
    class StackNode{
        public int data ; 
        public StackNode next ; 

        public StackNode(int data){
            this.data = data ; 
            this.next = null ; 
        }
        public int getData() {return this.data ;}
        public void setData(int data) {this.data = data ;}

        public StackNode getNext() {return this.next ;}
        public void setNext(StackNode next) {this.next = next ;}

        public int key{
            get {return data ;}
            set {data = value ;}
        }
        public StackNode newNext{
            get {return next ;}
            set {next = value ;}
        }

        public bool isEmpty(StackNode node){
            return node == null ; 
        }

        public void push(StackNode node, int data){
            StackNode newNode = new StackNode(data) ; 
            if(node == null){
                node = newNode ;
            }else{
                StackNode temp = node ; 
                newNode.next = temp ; 
                node = newNext ; 
            }   
        }

        public int pop(StackNode node){
            if(isEmpty(node)){
                Console.WriteLine("Stack is underflow !!!") ; 
                return 0 ; 
            }

            StackNode temp = node ; 
            node = node.next ; 

            return temp.data ; 
        }

        public int peek(StackNode node){
            if(isEmpty(node)){
                Console.WriteLine("Stack is underflow !!!") ;
                return 0 ; 

            }
            return node.data ;  
        }
    }

    public class TextProgram{
        public static void Main(string[] args){
            StackNode stack = new StackNode(10) ; 
            stack.push(stack, 20) ; 
            stack.push(stack, 30) ; 
            stack.push(stack, 40) ;

            Console.WriteLine("Element current in stack : "+stack.peek(stack)) ;  
        }
    }
}